appinventor/blocklyeditor/src/blocks/lists.js

```diff
@@ -468,3 +467,1519 @@ Blockly.Blocks['lists_join_with_separator'] = {
+  },
+  blocksInScope: function() {
+    var testBlock = this.getInputTargetBlock('TEST');
+    if (testBlock) {
+      return [testBlock];
+    } else {
+      return [];
+    }
+  },
+  declaredNames: function() {
+    return [this.getFieldValue('VAR')];
+  },
+  renameVar: function(oldName, newName) {
+    if (Blockly.Names.equals(oldName, this.getFieldValue('VAR'))) {
+      this.setFieldValue(newName, 'VAR');
+    }
+  },
+  typeblock: [{ translatedName: Blockly.Msg.LANG_LISTS_FILTER_INPUT_COLLAPSED_TEXT }]
+};
+
+
+Blockly.Blocks['lists_reduce'] = {
+  // For each loop.
+  category : 'Lists',
+  helpUrl : Blockly.Msg.LANG_LISTS_REDUCE_HELPURL,
+  init : function() {
+    this.setColour(Blockly.LIST_CATEGORY_HUE);
+    this.appendValueInput('LIST')
+        .setCheck(Blockly.Blocks.Utilities.YailTypeToBlocklyType("list",Blockly.Blocks.Utilities.INPUT))
+        .appendField(Blockly.Msg.LANG_LISTS_REDUCE_TITLE_REDUCE)
+        .appendField(Blockly.Msg.LANG_LISTS_REDUCE_INPUT_INLIST)
+        .setAlign(Blockly.ALIGN_RIGHT);
+    this.appendValueInput('INITANSWER')
+        .appendField(Blockly.Msg.LANG_LISTS_REDUCE_INPUT_INITIAL_ANSWER)
+        .setAlign(Blockly.ALIGN_RIGHT);
+    this.appendDummyInput('DESCRIPTION')
+        .appendField(Blockly.Msg.LANG_LISTS_REDUCE_INPUT_COMBINE)
+        .appendField(new Blockly.FieldParameterFlydown(Blockly.Msg.LANG_LISTS_REDUCE_INPUT_VAR,
+                                                       true, // name is editable
+                                                       Blockly.FieldFlydown.DISPLAY_BELOW),
+                     'VAR1')
+        .appendField(Blockly.Msg.LANG_LISTS_REDUCE_INPUT_AND)
+        .appendField(new Blockly.FieldParameterFlydown(Blockly.Msg.LANG_LISTS_REDUCE_INPUT_ANSWER,
+                                                       true, // name is editable
+                                                       Blockly.FieldFlydown.DISPLAY_BELOW),
+                     'VAR2')
+        .setAlign(Blockly.ALIGN_RIGHT);
+    this.appendIndentedValueInput('COMBINE');
+    this.setOutput(true, null);
+    this.setTooltip(Blockly.Msg.LANG_LISTS_REDUCE_TOOLTIP);
+  },
+  getVars: function() {
+    var names = []
+    names.push(this.getFieldValue('VAR1'));
+    names.push(this.getFieldValue('VAR2'));
+    return names;
+  },
+  blocksInScope: function() {
+    var combineBlock = this.getInputTargetBlock('COMBINE');
+    if (combineBlock) {
+      return [combineBlock];
+    } else {
+      return [];
+    }
+  },
+  declaredNames: function() {
+    return this.getVars();
+  },
+  renameVar: function(oldName, newName) {
+    if (Blockly.Names.equals(oldName, this.getFieldValue('VAR1'))) {
+      this.setFieldValue(newName, 'VAR1');
+    }
+    if (Blockly.Names.equals(oldName, this.getFieldValue('VAR2'))) {
+      this.setFieldValue(newName, 'VAR2');
+    }
+  },
+  typeblock: [{ translatedName: Blockly.Msg.LANG_LISTS_REDUCE_TITLE_REDUCE }]
+};
+
+Blockly.Blocks['lists_reverse'] = {
```

Comment:
For now we want to keep the original 'lists_reverse' blocks (see line 352, above), so please comment out this block's entire definition.

Reply:
Fixed